test/framework/src/main/java/org/opensearch/test/InternalTestCluster.java

```diff
@@ -2770,6 +2770,7 @@ public synchronized void assertAfterTest() throws Exception {
                 try {
                     env.shardLock(id, "InternalTestCluster assert after test", TimeUnit.SECONDS.toMillis(5)).close();
                 } catch (ShardLockObtainFailedException ex) {
+                    logger.error("Obtained shard lock failed", ex);
```

Comment:
The below fail function should effectively fail the tests with the needed logging already?

Reply:
fail function will log something like `java.lang.AssertionError: Shard [.plugins-ml-config][0] is still locked after 5 sec waiting`, it will have shard information, but lack of why it still been locked and who hold the lock? that's the main purpose of add this log. The exception will have these information. 

> org.opensearch.env.ShardLockObtainFailedException: [.plugins-ml-config][0]: obtaining shard lock for [InternalTestCluster assert after test] timed out after [5000ms], lock already held for [starting shard] with age [20365ms]

Reply:
Maybe improve the fail logging instead?

Reply:
sounds good, changed to improve existing fail assertion message.